flexx/event/_handler.py

```diff
@@ -375,22 +375,40 @@ def _connect_to_event(self, index):
         """
         connection = self._connections[index]
 
-        # Disconnect
-        while len(connection.objects):
-            ob, type = connection.objects.pop(0)
-            ob.disconnect(type, self)
-
+        # Prepare disconnecting
+        old_objects = connection.objects  # (ob, type) tuples
+        connection.objects = []
+        
         # Obtain root object and setup connections
         ob = self._ob1()
         if ob is not None:
             self._seek_event_object(index, connection.parts, ob)
-
+        new_objects = connection.objects
```

Comment:
if `connection.objects` returns a reference, your code might not do what you think it does

Reply:
That's why there is a `connection.objects = []` a few lines above :)  (it's a simple attribute, not a property)

Reply:
okay.